scarlet-websocket-okhttp/src/main/java/com/tinder/scarlet/websocket/okhttp/OkHttpClientUtils.kt

```diff
@@ -16,5 +18,16 @@ fun OkHttpClient.newWebSocketFactory(requestFactory: RequestFactory): WebSocket.
 }
 
 fun OkHttpClient.newWebSocketFactory(url: String): WebSocket.Factory {
+    if (url.startsWith("ws://")) {
```

Comment:
dont think this check will be necessary

Reply:
We need this check because people using SSL don't need to have cleartext traffic permitted. OkHttp does this check with the combination of code [here](https://github.com/square/okhttp/blob/20cd3a08935c78c75dae015964608b2c233d629d/okhttp/src/main/java/okhttp3/Request.kt#L175), [here](https://github.com/square/okhttp/blob/7a641531c8e03b10dc4fbf9f2465775d7ecc7649/okhttp/src/main/java/okhttp3/Address.kt#L75) and [here](https://github.com/square/okhttp/blob/7a641531c8e03b10dc4fbf9f2465775d7ecc7649/okhttp/src/main/java/okhttp3/internal/connection/RealConnection.kt#L154). I have added the "ignoreCase = true" parameter to String.startsWith that OkHttp uses.